core/src/main/scala/org/apache/spark/resource/ResourceProfileBuilder.scala

```diff
@@ -31,7 +31,7 @@ import org.apache.spark.annotation.Evolving
  * requirements between stages.
  */
 @Evolving
-class ResourceProfileBuilder() {
+private [spark] class ResourceProfileBuilder() {
```

Comment:
Shall we remove `@Evolving` if we mark this as `private [spark]`?

Reply:
BTW, is there any reason to make this `private` in this PR?

Reply:
the intention was to be private until rest of stage level scheduling done and just remove private, so would rather not remove evolving, but you are right this really should be separated out as its not necessarily related, I just noticed it missing, I can separate it out.